### DISCUSSION TITLE
Why does it seem HISTSIZE and SAVEHIST are backwards?

### DISCUSSION THREAD

**Discussion by ghost** [ohmyzsh/ohmyzsh#9833] [Help] [answered]
HISTSIZE=50000
SAVEHIST=10000

If HISTSIZE is how many records are loaded into memory, and SAVEHIST is how many records are saved to file, 
Why would we set the records to be loaded into memory higher than what will ever be in the file?

Shouldn't these two values be swapped?

**Comment by rwmitchell** [comment, marked as answer]
No

histsize is how many entries are in memory. Can’t save more than you have.

**Reply by ghost** [reply]
Sorry, this still isn't making sense, shouldn't they at least be equal?

I don't understand the point of having the buffer be 5x the size of what will ever be in the history file

**Reply by rwmitchell** [reply]
It's a default setting, change it to what makes you happy.

I don't close my sessions, so they get quite large, but when I do, I don't need to save everything. You do you.

**Reply by ghost** [reply]
I get that :) just wanted to understand the use case. Thanks!

**Reply by rwmitchell** [reply]
You can't save more than you have in memory, so having them inverted would be useless.
This way you can have a larger history than you save, if you want.  If not, make them the same.

**Reply by mattalxndr** [reply]
> I don't close my sessions, so they get quite large

@rwmitchell You might be interested in `setopt INC_APPEND_HISTORY`, which causes entries to be flushed to disk as they are entered, as opposed to waiting until you exit the session. 

It's nice to open a new prompt and have access to the command you just entered into your last one. 

Frankly, I'm not sure why it's not enabled by default. I can only assume they wanted to retain that good old bashy feeling.
